# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../strace/mips64.patch
# Copyright (C) 2010 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Fixes:
syscall.c: In function 'syscall_enter':
syscall.c:2038: error: 'pid' undeclared (first use in this function)

  - Jan Rovins <janr@adax.com>

diff -ur strace-4.5.19/syscall.c strace-4.5.19.fix/syscall.c
--- strace-4.5.19/syscall.c	2009-10-12 15:54:21.000000000 -0400
+++ strace-4.5.19.fix/syscall.c	2010-03-17 14:47:17.000000000 -0400
@@ -2035,7 +2035,7 @@
		else
			nargs = tcp->u_nargs = MAX_ARGS;

-		if (ptrace (PTRACE_GETREGS, pid, NULL, (long) &regs) < 0)
+		if (ptrace (PTRACE_GETREGS, tcp->pid, NULL, (long) &regs) < 0)
			return -1;

		for(i = 0; i < nargs; i++) {9
